### DISCUSSION TITLE
Unit of Measure (QuantityCodes)

### DISCUSSION THREAD

**Discussion by CommanderRaiker** [stephanstapel/ZUGFeRD-csharp#243] [Q&A] [answered]
Hello,

It appears that the provided QuantityCodes are incorrect or may be sourced from SAP. According to the xrechnung documentation, the quantity unit of measure code (BT-130) should be selected from the UN/ECE list.

You can refer to the list here: https://www.xrepository.de/details/urn:xoev-de:kosit:codeliste:rec20_3

Best regards,
Raik

**Comment by stephanstapel** [comment]
Hi Raik,
not quite. I decided against importing the full list since this vast amount of possibilities will result into significantly worse usability of the library. If you are missing specific values, feel free to add them here or issue a pull request (for the necessary codes)
Regards,
Stephan

**Reply by CommanderRaiker** [reply]
Ah, okay. Thank you for the prompt response. The issue is not that something is missing but rather that something does not conform to UN/ECE standards. I cannot find the following codes in the UN/ECE list: HAR, NPR, and all codes starting with X.
Best,
Raik

**Reply by stephanstapel** [reply, marked as answer]
good point. I will mark HAR and NPR as obsolete!
X* was defined here:
[documentation/zugferd22de/Documentation/EN16931 code lists values v8 - used from 2021-09-06.xlsx](https://github.com/stephanstapel/ZUGFeRD-csharp/blob/050043411b74fb559b90671a28fd99e4761682e8/documentation/zugferd22de/Documentation/EN16931%20code%20lists%20values%20v8%20-%20used%20from%202021-09-06.xlsx)
and it looks like they still are (using XPR as an example):
https://docs.peppol.eu/poacc/billing/3.0/2023-Q2/codelist/UNECERec20/

**Comment by CommanderRaiker** [comment]
> X* was defined here: [documentation/zugferd22de/Documentation/EN16931 code lists values v8 - used from 2021-09-06.xlsx](https://github.com/stephanstapel/ZUGFeRD-csharp/blob/050043411b74fb559b90671a28fd99e4761682e8/documentation/zugferd22de/Documentation/EN16931%20code%20lists%20values%20v8%20-%20used%20from%202021-09-06.xlsx) 

Thank you for the tip. I couldn't find that during my research.
Best,
Raik

**Comment by stephanstapel** [comment]
I understand why XRechnung simply suggests the full list of codes. Not only as a maintainer and user of a library like ZUGFeRD-sharp but also as someone who would need to impoort data into an ERP system, such a long list is a nightmare. Imagine receiving invoices from various senders and having to map each and any code...

**Comment by CommanderRaiker** [comment]
Hello. After reviewing our unit codes in the ERP system, I've identified a few additional units to include. I would be grateful if you could incorporate them in the next update.
Best regards
Raik

```
,

/// <summary>
/// Rolle
/// </summary>
XRO,

/// <summary>
/// Dose
/// </summary>
XTN,

/// <summary>
/// Kanister
/// </summary>
XCI,

/// <summary>
/// Tube
/// </summary>
XTU,

/// <summary>
/// Beutel
/// </summary>
XBG,

/// <summary>
/// (Papier) Bogen
/// </summary>
XST,

/// <summary>
/// Sack
/// </summary>
XSA,

/// <summary>
/// Fass
/// </summary>
XBA,

/// <summary>
/// Eimer
/// </summary>
XBJ,

/// <summary>
/// Gramm
/// </summary>
GRM,

/// <summary>
/// Kit
/// </summary>
KT

```

**Comment by stephanstapel** [comment]
done:
https://github.com/stephanstapel/ZUGFeRD-csharp/commit/744ad6a0690781b3c2ceca6b15f2261912608779
